internal/expr/query.go

```diff
@@ -87,19 +87,22 @@ func (pe *PreparedExpr) Query(args ...any) (ce *QueryExpr, err error) {
 				return nil, fmt.Errorf(`map %q does not contain key %q`, outerType.Name(), tm.name)
 			}
 		}
-		qargs = append(qargs, sql.Named("sqlair_"+strconv.Itoa(i), val.Interface()))
+		params = append(params, sql.Named("sqlair_"+strconv.Itoa(i), val.Interface()))
 	}
-	return &QueryExpr{outputs: pe.outputs, sql: pe.sql, args: qargs}, nil
+	return &QueryValues{outputs: te.outputs, sql: te.sql, params: params}, nil
 }
 
 var scannerInterface = reflect.TypeOf((*sql.Scanner)(nil)).Elem()
 
-// ScanArgs returns list of pointers to the struct fields that are listed in qe.outputs.
-// All the structs and maps mentioned in the query must be in outputArgs.
-func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) (scanArgs []any, onSuccess func(), err error) {
+// ScanArgs takes the result column names and the SQLair output arguments. It
+// returns list of pointers to be used as arguments for rows.Scan. They point to
+// locations within the SQLair output arguments. The onSuccess function should
+// be run once the pointers have been scanned to populate output arguments that
+// cannot be written to directly (such as map values).
```

Comment:
I think this comment can be improved a little if we reword it to avoid saying "arguments" for two different things, and if we avoid talking about the implementation (i.e. all we care about is that `onSuccess` needs to be ran after rows.Scan, we do not care why in this comment).

What do you think of the following?
```
// ScanArgs produces a list of pointers to be passed to rows.Scan. After a
// successful call, the onSuccess function must be invoked. The end result is
// that the outputArgs will be populated with the results from the query.
// All the structs/maps/slices mentioned in the query must be in outputArgs.
```
plus changing `columns` to `columnNames`.

Reply:
Fair enough, the previous comment was a bit rambly, I've changed it to a more concise version of your suggestion.